.github/workflows/maven.yml

```diff
@@ -124,3 +124,10 @@ jobs:
       - name: Running integration tests
         shell: bash
         run: mvn install -e -B -V -Prun-its,embedded -DmavenDistro="$GITHUB_WORKSPACE/maven/apache-maven/target/apache-maven-bin.zip" -f maven-integration-testing/pom.xml
+
+      - name: Upload artifact on integration testing
+        uses: actions/upload-artifact@v4
+        if: failure() && matrix.os != 'windows-latest'
```

Comment:
why skip on windows?

Reply:
Because the upload artifact component has a problem: https://github.com/actions/upload-artifact/issues/240